src/codeu/chat/server/Controller.java

```diff
@@ -34,9 +36,36 @@
   private final Model model;
   private final Uuid.Generator uuidGenerator;
 
-  public Controller(Uuid serverId, Model model) {
+  private FileWriter fileWriter;
+  private boolean loading;
+
+  public Controller(Uuid serverId, Model model, FileWriter fileWriter) {
     this.model = model;
     this.uuidGenerator = new RandomUuidGenerator(serverId, System.currentTimeMillis());
+
+    // store the FileWriter to user
+    this.fileWriter = fileWriter;
+    this.loading = true;
+    startUp();
+    this.loading = false;
+  }
+
+  public void startUp() {
+    FileLoader fileLoader = new FileLoader(this);
+    fileLoader.loadState();
+    new Thread(fileWriter).start();
+  }
+
+  // save the state from this Writeable to transaction log file
+  private void save(Writeable x) {
+    if(fileWriter == null)
+      return;
+    try {
+      if(!loading)
```

Comment:
Would this actually be an issue? Are you worried that a save might occur between where startUp() returns and when you reset this.loading in the constructor? If so, it's probably a good idea to start the fileWriter thread after you set this.loading to false in the constructor.